rnote-ui/src/workspacebrowser/filerow/mod.rs

```diff
@@ -200,100 +200,11 @@ impl FileRow {
-
-                            let parent_path_1 = parent_path.clone();
-                            rename_entry.connect_text_notify(clone!(@weak rename_apply_button => move |rename_entry| {
-                                let new_file_path = parent_path_1.join(rename_entry.text().to_string());
-                                let new_file = gio::File::for_path(new_file_path);
-
-                                // Disable apply button to prevent overwrites when file already exists
-                                rename_apply_button.set_sensitive(!new_file.query_exists(None::<&gio::Cancellable>));
-                            }));
-
-                            rename_cancel_button.connect_clicked(clone!(@weak rename_popover => move |_| {
-                                rename_popover.popdown();
-                            }));
-
-                            rename_apply_button.connect_clicked(clone!(@weak rename_popover, @weak rename_entry => move |_| {
-                                let new_file_path = parent_path.join(rename_entry.text().to_string());
-                                let new_file = gio::File::for_path(new_file_path);
-
-                                if new_file.query_exists(None::<&gio::Cancellable>) {
-                                    // Should have been caught earlier, but making sure
-                                    log::error!("file already exists");
-                                } else {
-                                    if let Err(e) = current_file.move_(&new_file, gio::FileCopyFlags::NONE, None::<&gio::Cancellable>, None) {
-                                        log::error!("rename file failed with Err {}", e);
-                                    }
-
-                                    rename_popover.popdown();
-                                }
-                            }));
-
-                            rename_popover.popup();
-                        }
-                    }
-                }
-            }));
-
-        // Trash file
-        action_trash_file.connect_activate(clone!(@weak self as filerow => move |_action_trash_file, _| {
-                if let Some(current_file) = filerow.current_file() {
-                    current_file.trash_async(glib::PRIORITY_DEFAULT, None::<&gio::Cancellable>, clone!(@weak filerow => move |res| {
-                        if let Err(e) = res {
-                            log::error!("filerow trash file failed with Err {}", e);
-                        } else {
-                            filerow.set_current_file(None);
-                        }
-                    }));
-                }
-            }));
+        self.imp().action_group.add_action(&self.open_action(&appwindow));
+        self.imp().action_group.add_action(&self.rename_action());
+        self.imp().action_group.add_action(&self.trash_action());
+        self.imp().action_group.add_action(&self.duplicate_action(&appwindow));
```

Comment:
Would you prefer to have it like this?
```rust
self.imp().action_group.add_action(&actions::open_action(self, &appwindow));
self.imp().action_group.add_action(&actions::rename_action(self));
self.imp().action_group.add_action(&actions::rash_action(self);
self.imp().action_group.add_action(&actions::duplicate_action(self, &appwindow));
```

Reply:
first one is fine, but please rename them to `create_open_action`, `create_rename_action` etc.

Reply:
what do you mean with "first"? The first from the current code or the first of my suggestion?

Reply:
current code. so `self.create_`